app/objects/c_agent.py

```diff
@@ -50,10 +51,11 @@ def store(self, ram):
             ram['agents'].append(self)
             return self.retrieve(ram['agents'], self.unique)
         else:
+            now = datetime.now()
             existing.update('trusted', self.trusted)
             if existing.trusted:
-                existing.update('last_trusted_seen', datetime.now())
-            existing.update('last_seen', datetime.now())
+                existing.update('last_trusted_seen', now)
+            existing.update('last_seen', now)
```

Comment:
shouldn't both of these come from properties, self.last_seen and self.last_trusted_seen

Reply:
this looks like the only spot where we update the last_seen values, but it does happen in kind of a weird implicit way / under the hood way.  We could move the setting the timestamps to https://github.com/mitre/caldera/blob/master/app/objects/c_c2.py#L26 and then make these fields work like other fields.

Reply:
I would suggest merging this as-is and circling back to the timestamps in general. a variety of objects use timestamps (on creation and/or updates) and if we were to make a change, we could do something more robust with them in mind. @wbooth thoughts?

Reply:
yep, we talked about it and this is good as is, forgot to put approval on the PR. this is good to go!